sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -186,26 +190,112 @@ private SdkFunctionMetadata CreateSdkFunctionMetadata(string functionName, strin
         {
+            if (!TryAddOutputBindingFromMethod(bindingMetadata, method))
+            {
+                AddOutputBindingsFromReturnType(module, bindingMetadata, method);
+            }
+        }
+
+        private void AddOutputBindingsFromReturnType(ModuleDefinition module, IList<ExpandoObject> bindingMetadata, MethodDefinition method)
+        {
+            string? returnType = GetTaskElementType(method.ReturnType);
+
+            if (returnType is not null)
+            {
+                if (string.Equals(returnType, HttpResponseType))
+                {
+                    AddHttpOutputBinding(bindingMetadata, ReturnBindingName);
+                }
+                else
+                {
+                    TypeDefinition returnDefinition = module.GetType(returnType)
+                        ?? throw new InvalidOperationException($"Couldn't find the type definition {returnType}");
+
+                    AddOutputBindingsFromProperties(bindingMetadata, returnDefinition);
+                }
+            }
+        }
+
+        private void AddOutputBindingsFromProperties(IList<ExpandoObject> bindingMetadata, TypeDefinition typeDefinition)
+        {
+            bool foundHttpOutput = false;
+
+            foreach (PropertyDefinition property in typeDefinition.Properties)
+            {
+                if (string.Equals(property.PropertyType.FullName, HttpResponseType))
+                {
+                    if (foundHttpOutput)
+                    {
+                        throw new InvalidOperationException($"Found multiple properties with type '{HttpResponseType}' " +
+                            $"inside type '{typeDefinition.FullName}' used for output binding." +
+                            $"You can only have one Http Output Binding type in your return data definition.");
+                    }
+
+                    foundHttpOutput = true;
+                    AddHttpOutputBinding(bindingMetadata, property.Name);
+                    continue;
+                }
+
+                AddOutputBindingFromProperty(bindingMetadata, property);
+            }
+        }
+
+        private void AddOutputBindingFromProperty(IList<ExpandoObject> bindingMetadata, PropertyDefinition property)
+        {
+            bool foundOutputAttribute = false;
+
+            foreach (CustomAttribute propertyAttribute in property.CustomAttributes)
+            {
+                if (IsOutputBindingType(propertyAttribute))
+                {
+                    if (foundOutputAttribute)
+                    {
+                        throw new InvalidOperationException($"Found multiple output attritbues on property '{property.Name}' inside the function return type." +
+                                    $"You can only have one Http Output Binding type in your return data definition.");
+                    }
```

Comment:
nit (includes spelling and accurate content):
$"Found multiple output attributes on property '{property.Name}' defined in the function return type '{typeName}'. Only one output binding attribute is is supported on a property."